addons/link_tracker/controller/main.py

```diff
@@ -5,15 +5,23 @@
 from odoo.http import request
 
 
+ignore_user_agent_prefix = ["twitterbot", "facebookexternalhit", "facebot",
+                            "LinkedInBot", "WhatsApp", "Googlebot", "Pinterest", "Instagram"]
+
+
 class LinkTracker(http.Controller):
 
     @http.route('/r/<string:code>', type='http', auth='public', website=True)
     def full_url_redirect(self, code, **post):
-        country_code = request.session.geoip and request.session.geoip.get('country_code') or False
-        request.env['link.tracker.click'].sudo().add_click(
-            code,
-            ip=request.httprequest.remote_addr,
-            country_code=country_code
-        )
+
+        user_agent = str(request.httprequest.user_agent) or ""
+        if not any(user_agent.lower().startswith(agent.lower()) for agent in ignore_user_agent_prefix):
```

Comment:
Can you change all `ignore_user_agent_prefix` in lower case, and just do `startswith(agent)` ?